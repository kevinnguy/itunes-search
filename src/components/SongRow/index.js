import React from 'react';
import { Image, StyleSheet, Text, View } from 'react-native';

export default function SongRow({ data }) {
  const { artistName, artworkUrl100, trackName, collectionName } = data;
  return (
    <View style={styles.container}>
      <Image style={styles.artwork} source={{ uri: artworkUrl100 }} />
      <View style={styles.details}>
        <Text style={styles.track} numberOfLines={1}>
          {trackName}
        </Text>
        <Text style={styles.artist} numberOfLines={1}>
          {artistName}
        </Text>
        <Text style={styles.collection} numberOfLines={2}>
          {collectionName}
        </Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    height: 120,
    padding: 10,
  },
  artwork: {
    width: 100,
  },
  details: {
    flex: 1,
    justifyContent: 'space-between',
    marginLeft: 10,
  },
  artist: {
    fontSize: 20,
  },
  track: {
    fontSize: 20,
  },
  collection: {
    fontSize: 18,
  },
});
