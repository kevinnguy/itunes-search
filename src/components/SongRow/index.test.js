import React from 'react';
import renderer from 'react-test-renderer';

import SongRow from './index';
import mockData from '../../../__tests__/mockData.json';

const rowData = mockData.results[0];

test('renders correctly with data', () => {
  const tree = renderer.create(<SongRow data={rowData} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders more details', () => {
  const tree = renderer.create(<SongRow data={rowData} showDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});
