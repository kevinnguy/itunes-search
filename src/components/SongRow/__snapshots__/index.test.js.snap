// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with data 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "height": 120,
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
      }
    }
    style={
      Array [
        Object {
          "marginRight": 10,
          "width": 100,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "justifyContent": "space-between",
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Better Together
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Jack Johnson
    </Text>
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      In Between Dreams (Bonus Track Version)
    </Text>
  </View>
</View>
`;

exports[`renders more details 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "row",
        "height": 120,
        "padding": 10,
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "https://is2-ssl.mzstatic.com/image/thumb/Music118/v4/24/46/97/24469731-f56f-29f6-67bd-53438f59ebcb/source/100x100bb.jpg",
      }
    }
    style={
      Array [
        Object {
          "marginRight": 10,
          "width": 100,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "justifyContent": "space-between",
        },
        false,
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Track: Better Together
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Artist: Jack Johnson
    </Text>
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Album: In Between Dreams (Bonus Track Version)
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      $1.29
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Buy the album for $11.99
    </Text>
    <Text
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Genre: Rock
    </Text>
  </View>
</View>
`;
